### DISCUSSION TITLE
Restreaming Youtube video from URL (Ant Media Server)

### DISCUSSION THREAD

**Discussion by hyalicom** [ant-media/Ant-Media-Server#5175] [Q&A]
Hi, I would like to ask, Is there any way to Restream Youtube Video, Instead of uploading Videos one by one to the VOD section, is there any way (to put URL of the Youtube Video) and stream that!

**Comment by USAMAWIZARD** [comment]
Hi,can you please clarify , Do you want to use youtube URL as stream src in Ant Media to live stream to youtube ?

**Comment by hyalicom** [comment]
Hi. Thanks for the response. Yes i actually want to stream from youtube
URL, Youtubr video as source

The main idea is, It will save my time needed to upload videos in the VOD
section

Thanks
Ali

**Comment by klasadoroty** [comment]
Hi.
On the Internet, anything is possible.
Get interested in this: streamline
With it you download streams from youtube and upload to a streaming server.
It works well as an SRT to the server.

**Reply by tremzo** [reply]
Can you explain what you mean by SRT to the server? Are you automatically downloading the Youtube video and streaming via SRT to Ant Media Server? 

Using the embed URL is impossible because Youtube prevents the URL from being executed, but please detail what workflow you are already using so we can investigate.

**Comment by klasadoroty** [comment]
Sorry, I rearranged the letters:
not streamline only STREAMLINK
Find streamlink, github. Read this.
"streaming via SRT to Ant Media Server" -yes, AMS or other
It is work

**Comment by mekya** [comment]
I have an idea for this problem. 
There is a WebPage Recording Plugin and you can give youtube URL to this plugin and it'll publish that page to the Ant Media Server. 

Even if It's not merged to the [Plugins master](https://github.com/ant-media/Plugins/) repository yet, fortunately, you can [get it here](https://github.com/ant-media/Plugins/tree/addWebPageDevelopmentPlugin/WebpageRecordingPlugin)

**Reply by tremzo** [reply]
@klasadoroty Yes, this is the answer. You can use the webpage recording plugin that can live stream a webpage. @mekya the only issue is that the link to the JAR file is reaching a 404 https://github.com/ant-media/Plugins/raw/master/WebpageRecordingPlugin/build/webpage-recording-plugin.jar what is the updated link?

**Reply by mekya** [reply]
It's available under the build directory -> https://github.com/ant-media/Plugins/tree/addWebPageDevelopmentPlugin/WebpageRecordingPlugin/build

The link will work when it's merged into the master

**Reply by klasadoroty** [reply]
Does this plugin heavily load the server?

**Reply by mekya** [reply]
This plugin will open a headless Chrome browser in the server and push the webpage to the server itself by WebRTC.  Yes, it has an overhead and I've not measured the overhead.

Btw, @USAMAWIZARD has recommended another solution.  Take a look at the solution below if it works for you.

**Comment by USAMAWIZARD** [comment]
@hyalicom 
Streaming youtube video without uploading it to Ant Media Server 
you can also use ffmpeg youtube dl like this 
```
python3 -m pip install --force-reinstall https://github.com/yt-dlp/yt-dlp/archive/master.tar.gz

ffmpeg -re -stream_loop -10 -i  "$(yt-dlp  -g -f best https://www.youtube.com/watch?v=xvFZjo5PgG0)" -c:v libx264 -x264-params keyint=80:scenecut=0  -preset fast -tune zerolatency -f mpegts  srt://AMS_IP:4200?streamid=WebRTCAppEE/stream11

https://AMS_IP:5443/WebRTCAppEE/play.html?id=stream11&playOrder=hls
```
